query-engine/connectors/query-connector/src/error.rs

```diff
@@ -9,10 +9,12 @@ use user_facing_errors::{query_engine::DatabaseConstraint, KnownError};
 #[derive(Debug, Error)]
 #[error("{}", kind)]
 pub struct ConnectorError {
-    /// An optional error already rendered for users in case the migration core does not handle it.
+    /// An optional error already rendered for users.
     pub user_facing_error: Option<KnownError>,
     /// The error information for internal use.
     pub kind: ErrorKind,
+    /// Whether an error is transient and should be retried.
+    pub transient: bool,
```

Comment:
We use this internal field to retry on transient errors while still forwarding the same errors to users. This field is _not_ forwarded to users though. This helps making this change not breaking for now.

Note: I tried adding a new kind of error and wrapping the root error but I haven't managed to cleanly unwrap the underlying error before sending it to the user. Mainly because `CoreError` is transformed into many different types of errors and it felt easy to miss one place.